src/platform/silabs/SiWx917/wifi/wfx_host_events.h

```diff
@@ -41,7 +41,10 @@
 #define SL_WFX_CONNECT_IND_ID (2)
 #define SL_WFX_DISCONNECT_IND_ID (3)
 #define SL_WFX_SCAN_COMPLETE_ID (4)
-#define WFX_RSI_SSID_SIZE (64)
+// MAX SSID LENGTH excluding NULL character
+#define WFX_MAX_SSID_LENGTH 32
+// MAX PASSKEY LENGTH including NULL character
```

Comment:
This is NOT including null character (and if fact is unrelated to null-termination)

Reply:
sl_wifi_constants.h has:

```
/// Max Length of Wi-Fi PSK credential
#define SL_WIFI_MAX_PSK_LENGTH 32
```

Be caseful wherever you use the WFX_MAX_PASSKEY_LENGTH not to assume it has space for a null terminator at the maximum passcode size!

Reply:
this is not as per spec, looking for a fix from Silabs SDK

Reply:
Where is the +1 ?

Reply:
yes, I was thinking of porting to what you suggested i.e. using `chip::Platform::CopyString` but unless the limits are in spec we would not really solve the problem.